plugins/usbdmx/AsyncUsbTransceiverBase.cpp

```diff
@@ -29,36 +31,68 @@ namespace usbdmx {
 
 using ola::usb::LibUsbAdaptor;
 
-namespace {
-
 /*
  * Called by libusb when the transfer completes.
  */
 #ifdef _WIN32
 __attribute__((__stdcall__))
 #endif  // _WIN32
-void AsyncCallback(struct libusb_transfer *transfer) {
+void AsyncTransferCallback(struct libusb_transfer *transfer) {
   AsyncUsbTransceiverBase *widget = reinterpret_cast<AsyncUsbTransceiverBase*>(
     transfer->user_data);
-  widget->TransferComplete(transfer);
+  widget->TransferCompleteInternal(transfer);
+}
+
+void AsyncUsbTransceiverBase::TransferCompleteInternal(
+  struct libusb_transfer *transfer) {
+  {
+    ola::thread::MutexLocker locker(&m_mutex);
+    inflight_set::iterator i = m_inflight.find(transfer);
+    if (i == m_inflight.end()) { // not found
+      OLA_WARN << "Mismatched libusb transfer: "
+               << transfer << " not in inflight set";
+      return;
+    }
+
+    m_inflight.erase(i);
```

Comment:
Would this be easier replacing with our ola::STLLookupAndRemove? http://docs.openlighting.org/ola/doc/latest/group__stl.html#ga7a896801adfcea56e630bd5007c135e1

Reply:
Apologies, wrong command, after looking more carefully ola::STLRemove will do what you want:
http://docs.openlighting.org/ola/doc/latest/group__stl.html#ga6bab5568a6d6ef14d4faccea9eddd41b

Or is there a risk of multiple equal transfers in the set (I assume they have a unique ID, but maybe not)?

Reply:
> Or is there a risk of multiple equal transfers in the set (I assume they have a unique ID, but maybe not)?

Reading more carefully, they have to be unique, so no risk there.

Reply:
Sorry to push back on this, but honestly I don't see STLRemove having any advantage over callind `.erase()` here. All this change would do is invalidate my testing and perform another unnecessary traversal to find the right element again. A combined lookup and remove would have cleaned up the code a bit but for sure, but just with just the removal I don't see the point.